src/classes/BDI_Donations.cls

```diff
@@ -644,6 +673,11 @@ public class BDI_Donations {
                     || dataImport.DonationImportStatus__c == Label.bdiMatchedApplyNewPayment) {
                 npe01__OppPayment__c payment = paymentsByDataImportId.get(dataImport.Id);
                 if (payment != null) {
+                    if (shouldSetDonationBatchNumber(payment, dataImport)) {
+                        payment.put(npe01__OppPayment__c.npsp__Batch_Number__c,
```

Comment:
I don't think we should include the "npsp__" namespace on the batch number field!